Examples/Stereo-Inertial/ILLIXR_stereo_inertial_euroc.cc

```diff
@@ -0,0 +1,254 @@
+             (!dataum->img0.has_value() && !dataum->img1.has_value()));
+      vImuMeas.push_back(ORB_p);
+
+#ifdef CV_HAS_METRICS
+      cv::metrics::setAccount(new std::string{std::to_string(iteration_no)});
+      iteration_no++;
+      if (iteration_no % 20 == 0) {
+         cv::metrics::dump();
+      }
+#else
+#warning \
+    "No OpenCV metrics available. Please recompile OpenCV from git clone --branch 3.4.6-instrumented https://github.com/ILLIXR/opencv/. (see install_deps.sh)"
+#endif
+
+      std::optional<cv::Mat *> imLeft = imu_cam_buffer->img0;
+      std::optional<cv::Mat *> imRight = imu_cam_buffer->img1;
+      cv::Mat imLeftRect, imRightRect;
+
+#ifdef COMPILEDWITHC11
+      std::chrono::steady_clock::time_point t_Start_Rect =
+          std::chrono::steady_clock::now();
+#else
+      std::chrono::monotonic_clock::time_point t_Start_Rect =
+          std::chrono::monotonic_clock::now();
+#endif
+      cv::remap(**imLeft, imLeftRect, M1l, M2l, cv::INTER_LINEAR);
+      cv::remap(**imRight, imRightRect, M1r, M2r, cv::INTER_LINEAR);
+
+#ifdef COMPILEDWITHC11
+      std::chrono::steady_clock::time_point t_End_Rect =
+          std::chrono::steady_clock::now();
+#else
+      std::chrono::monotonic_clock::time_point t_End_Rect =
+          std::chrono::monotonic_clock::now();
+#endif
+      double t_rect = std::chrono::duration_cast<std::chrono::duration<double>>(
+                          t_End_Rect - t_Start_Rect)
+                          .count();
+
+      double tframe = dataum->dataset_time;
+
+      // If there is not cam data this func call, break early
+      if (!dataum->img0.has_value() && !dataum->img1.has_value()) {
+         return;
+      } else {
+#ifdef COMPILEDWITHC11
+         std::chrono::steady_clock::time_point t1 =
+             std::chrono::steady_clock::now();
+#else
+         std::chrono::monotonic_clock::time_point t1 =
+             std::chrono::monotonic_clock::now();
+#endif
+
+         // Pass the images to the SLAM system
+         cv::Mat Tcw =
+             SLAM_sys->TrackStereo(imLeftRect, imRightRect, tframe, vImuMeas);
+
+#ifdef COMPILEDWITHC11
+         std::chrono::steady_clock::time_point t2 =
+             std::chrono::steady_clock::now();
+#else
+         std::chrono::monotonic_clock::time_point t2 =
+             std::chrono::monotonic_clock::now();
+#endif
+         vImuMeas.clear();
+         cv::Mat Rwc = Tcw.rowRange(0, 3).colRange(0, 3).t();
+         cv::Mat twc = -Rwc * Tcw.rowRange(0, 3).col(3);
+         Eigen::Vector3f position{twc.at<float>(0), twc.at<float>(1),
+                                  twc.at<float>(2)};
+         Eigen::Matrix3f emat3f;
+         cv2eigen(Rwc, emat3f);
+         Eigen::Quaternionf orientation(emat3f);
+         _m_pose->put(new pose_type{.sensor_time = imu_cam_buffer->time,
+                                    .position = position,
+                                    .orientation = orientation});
```

Comment:
This pose convertion part is wrong.
You need to figure out how ORB_SLAM3 encode poses.
This might be helpful but I failed to understand.
raulmur/ORB_SLAM2#428